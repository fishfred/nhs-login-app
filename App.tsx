/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */
import 'react-native-gesture-handler';

import React from 'react';

import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';

import {WelcomeScreen} from './components/screens/WelcomeScreen';
import {DashboardScreen} from './components/screens/DashboardScreen';

import {navigationRef} from './components/RootNavigation';
import {OpenidSettingsScreen} from './components/screens/OpenidSettingsScreen';
import {LoginWebviewScreen} from './components/screens/LoginWebviewScreen';
import {NhsLogin} from './components/NhsLogin';

import * as Colors from './styles/colors';
import {OpenIDDetailsScreen} from './components/screens/OpenidDetails';
import {EnvironmentScreen} from './components/screens/EnvironmentScreen';

import {RootStackParamList} from './services';
import {ChatScreen} from './components/screens/ChatScreen';

const Stack = createStackNavigator<RootStackParamList>();

declare const global: {HermesInternal: null | {}};

export default class App extends React.Component {
  render() {
    return (
      <>
        <NavigationContainer ref={navigationRef}>
          <Stack.Navigator
            headerMode="float"
            screenOptions={{
              headerTintColor: Colors.Blue,
              headerStyle: {
                backgroundColor: Colors.Blue,
              },
            }}>
            <Stack.Screen
              name="Welcome"
              component={WelcomeScreen}
              options={{
                header: WelcomeScreen.header,
              }}
            />
            <Stack.Screen
              name="Dashboard"
              component={DashboardScreen}
              options={{
                header: DashboardScreen.header,
              }}
            />
            <Stack.Screen
              name="OpenidSettings"
              component={OpenidSettingsScreen}
              options={{
                header: OpenidSettingsScreen.header,
              }}
            />
            <Stack.Screen
              name="OpenidDetails"
              component={OpenIDDetailsScreen}
              options={{
                header: OpenIDDetailsScreen.header,
              }}
            />
            <Stack.Screen
              name="Environment"
              component={EnvironmentScreen}
              options={{
                header: EnvironmentScreen.header,
              }}
            />
            <Stack.Screen
              name="LoginWebview"
              component={LoginWebviewScreen}
              options={{
                header: LoginWebviewScreen.header,
              }}
            />
            <Stack.Screen
              name="Chat"
              component={ChatScreen}
              options={{
                header: ChatScreen.header,
              }}
            />
          </Stack.Navigator>
        </NavigationContainer>
      </>
    );
  }
}
