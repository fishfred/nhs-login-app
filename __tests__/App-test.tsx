/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer, { act } from 'react-test-renderer';

it('renders correctly', () => {
  let root: renderer.ReactTestRenderer;
  act(()=>{
    root = renderer.create(<App />);
  });

  //@ts-ignore
  expect(root).toMatchSnapshot();
  //@ts-ignore
  root.unmount();
});
