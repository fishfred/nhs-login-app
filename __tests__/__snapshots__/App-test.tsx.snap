// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#005eb8",
                "flexDirection": "row",
                "minHeight": 60,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessibilityLabel="Nhs Login App Welcome Screen"
                accessibilityRole="header"
              >
                <SvgMock
                  fill="#ffffff"
                  height={40}
                  width={60}
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "display": "flex",
                              "flex": 1,
                              "flexDirection": "column-reverse",
                              "marginBottom": 40,
                              "marginHorizontal": 20,
                            }
                          }
                        >
                          <View
                            accessibilityLabel="Continue with NHS Login"
                            accessibilityRole="button"
                            accessible={true}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                          >
                            <View
                              style={
                                Object {
                                  "alignSelf": "flex-end",
                                  "display": "flex",
                                  "height": 60,
                                  "marginTop": 10,
                                  "width": "100%",
                                }
                              }
                            >
                              <SvgMock
                                style={
                                  Object {
                                    "alignSelf": "flex-end",
                                    "display": "flex",
                                    "height": 60,
                                    "marginTop": 10,
                                    "width": "100%",
                                  }
                                }
                              />
                            </View>
                          </View>
                          <View
                            accessibilityLabel="Modify Scopes"
                            accessibilityRole="button"
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "height": 57,
                                "marginTop": 5,
                              }
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "height": 53,
                                    "justifyContent": "center",
                                    "left": 0,
                                    "position": "absolute",
                                    "top": 0,
                                    "width": "100%",
                                    "zIndex": -1,
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#414141",
                                    "borderRadius": 4,
                                    "height": 53,
                                    "justifyContent": "center",
                                    "left": 0,
                                    "marginTop": 4,
                                    "position": "absolute",
                                    "top": 0,
                                    "width": "100%",
                                    "zIndex": 3,
                                  }
                                }
                              />
                            </View>
                            <View
                              style={
                                Object {
                                  "backgroundColor": "#4c6272",
                                  "borderRadius": 4,
                                  "height": 53,
                                  "marginTop": 0,
                                  "padding": 14,
                                  "textTransform": "none",
                                  "width": "100%",
                                }
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "#ffffff",
                                    "fontSize": 19,
                                    "fontWeight": "bold",
                                    "textAlign": "center",
                                  }
                                }
                              >
                                Modify Scopes
                              </Text>
                            </View>
                          </View>
                          <View
                            accessibilityLabel="Clear Data"
                            accessibilityRole="button"
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "height": 57,
                                "marginTop": 5,
                              }
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "height": 53,
                                    "justifyContent": "center",
                                    "left": 0,
                                    "position": "absolute",
                                    "top": 0,
                                    "width": "100%",
                                    "zIndex": -1,
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#414141",
                                    "borderRadius": 4,
                                    "height": 53,
                                    "justifyContent": "center",
                                    "left": 0,
                                    "marginTop": 4,
                                    "position": "absolute",
                                    "top": 0,
                                    "width": "100%",
                                    "zIndex": 3,
                                  }
                                }
                              />
                            </View>
                            <View
                              style={
                                Object {
                                  "backgroundColor": "#4c6272",
                                  "borderRadius": 4,
                                  "height": 53,
                                  "marginTop": 0,
                                  "padding": 14,
                                  "textTransform": "none",
                                  "width": "100%",
                                }
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "#ffffff",
                                    "fontSize": 19,
                                    "fontWeight": "bold",
                                    "textAlign": "center",
                                  }
                                }
                              >
                                Clear Data
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
